### PR TITLE
Issue#348 image cropper with react image crop

### PR DESCRIPTION
**Pull request by runarfu**
GlobalDigitalLibraryio/issues#348

Far from perfect, but it does the job. Candidate for merge and deploy to test, in order to use it before further development.

### CODE REVIEW

File: packages/admin-frontend/package.json [hunk 1]
```diff
@@ -8,6 +8,7 @@
     "start": "NODE_ENV=production node server"
   },
   "dependencies": {
+    "@zeit/next-css": "^0.2.0",
     "@zeit/next-source-maps": "^0.0.2",
```
Comment (alexanbj) [inline packages/admin-frontend/package.json]: Remove the `@zeit/next-source-maps`dependency (currently not used)?

File: packages/admin-frontend/pages/crop.js [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+      rawImageQueryParameters: {
+        cropStartX: pcnt.cropStartX,
+        cropEndX: pcnt.cropEndX,
+        cropStartY: pcnt.cropStartY,
+        cropEndY: pcnt.cropEndY
+      }
+    };
+  };
+
+  _crop() {
```
Comment (alexanbj) [inline packages/admin-frontend/pages/crop.js]: Curious as to why this method is "marked private" as opposed to the other methods

File: packages/admin-frontend/pages/crop.js [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+    }
+  };
+
+  getExistingParameters = async () => {
+    const imageUrl =
+      this.props.imageUrl &&
+      this.props.imageUrl.substr(this.props.imageUrl.lastIndexOf('/'));
+    const url =
+      imageUrl &&
+      `${imageApiUrl}/images/stored-parameters${imageUrl.toString()}`;
```
Comment (alexanbj) [inline packages/admin-frontend/pages/crop.js]: is toString() necessary here?

File: packages/admin-frontend/pages/crop.js [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+      }
+    };
+  };
+
+  _crop() {
+    var pcnt = this.toPercentages(this.refs.cropper);
+    this.setState({ imageApiBody: this.toImageApiBody(pcnt) });
+  }
+
+  postToImageApi = async _ => {
```
[the remark this answers is not in the document]
Comment (runarfu) [reply packages/admin-frontend/pages/crop.js]: But it has to take an argument (the click event), and this is the way to ignore it?

File: packages/admin-frontend/pages/crop.js (hunk 2 above)
Comment (runarfu) [reply packages/admin-frontend/pages/crop.js]: No reason. Fixed.

File: packages/admin-frontend/pages/crop.js (hunk 3 above)
Comment (runarfu) [reply packages/admin-frontend/pages/crop.js]: No, I misunderstood a Flow error I got earlier.

File: packages/admin-frontend/pages/crop.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (runarfu) [reply packages/admin-frontend/pages/crop.js]: Got it.

File: packages/admin-frontend/pages/crop.js [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+    if (this.state.imageApiBody !== null) {
+      this.setState({ postResult: 'Posting…' });
+      const response = await fetch(`${imageApiUrl}/images/stored-parameters`, {
+        method: 'POST',
+        headers: {
+          Authorization: 'Bearer ' + getTokenFromLocalCookie(),
+          Accept: 'application/json'
+        },
+        body: JSON.stringify(this.state.imageApiBody)
+      });
```
[the remark this answers is not in the document]
Comment (runarfu) [reply packages/admin-frontend/pages/crop.js]: Will do this in a rewrite in the next version.

**Comment by runarfu** [comment]
@alexanbj Se over igjen?

**Comment by runarfu** [comment]
## TODO
- [ ] Release.
- [ ] Deploy to test.
- [ ] Deploy to staging.
- [ ] Deploy to prod.

**Pull request merged**